src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/VariantArray.cs

```diff
@@ -49,11 +50,15 @@ internal static class VariantArray
         // (guarenteed less than 28, in practice 0-2)
         private static readonly List<Type> s_generatedTypes = new List<Type>(0);
 
+        [RequiresUnreferencedCode("Types and members might be removed")]
         internal static MemberExpression GetStructField(ParameterExpression variantArray, int field)
         {
             return Expression.Field(variantArray, "Element" + field);
         }
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(Variant))]
```

Comment:
I'm not sure I understand what this `DynamicDependency` is for.

The `IL2055` message is warning about the generic type, in this case `VariantArray3<T>`. Since the trimmer doesn't know the type, it can't tell if the `T` argument is annotated with trimming attributes like `[DynamicallyAccessedMembers]`, so the trimmer is warning.

We can simply suppress the warning because the type is generated without any trimming annotations applied to the `T` argument.

Reply:
My reasoning here was following:
As you said `VariantArray{1,3,4,8}` are generated so they don't need anythings special but all of them depend on `Variant` (it's used as their field) which I wasn't sure that trimmer can detect so I added it as a dynamic dependency